cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ReplicationThrottleHelper.java

```diff
@@ -139,180 +156,208 @@ private boolean throttlingEnabled() {
+  private void setThrottledRateIfUnset(int brokerId) throws ExecutionException, InterruptedException {
     if (_throttleRate == null) {
       throw new IllegalStateException("Throttle rate cannot be null");
     }
-    String replicaThrottleRateConfigKey = throttleLeaderReplicaRate ? LEADER_THROTTLED_RATE : FOLLOWER_THROTTLED_RATE;
-    Properties brokerConfigs = _kafkaZkClient.getEntityConfigs(ConfigType.Broker(), String.valueOf(brokerId));
-    Object currThrottleRate = brokerConfigs.setProperty(replicaThrottleRateConfigKey, String.valueOf(_throttleRate));
-    if (currThrottleRate == null) {
-      LOG.debug("Setting {} to {} bytes/second for broker {}", replicaThrottleRateConfigKey, _throttleRate, brokerId);
-      ExecutorUtils.changeBrokerConfig(_adminZkClient, brokerId, brokerConfigs);
-    } else {
-      LOG.debug("Not setting {} for broker {} because pre-existing throttle of {} was already set",
-          replicaThrottleRateConfigKey, brokerId, currThrottleRate);
+    Config brokerConfigs = getBrokerConfigs(brokerId);
+    List<AlterConfigOp> ops = new ArrayList<>();
+    for (String replicaThrottleRateConfigKey : Arrays.asList(LEADER_THROTTLED_RATE, FOLLOWER_THROTTLED_RATE)) {
+      ConfigEntry currThrottleRate = brokerConfigs.get(replicaThrottleRateConfigKey);
+      if (currThrottleRate == null) {
+        LOG.debug("Setting {} to {} bytes/second for broker {}", replicaThrottleRateConfigKey, _throttleRate, brokerId);
+       ops.add(new AlterConfigOp(new ConfigEntry(replicaThrottleRateConfigKey, String.valueOf(_throttleRate)), AlterConfigOp.OpType.SET));
+      } else {
+        LOG.debug("Not setting {} for broker {} because pre-existing throttle of {} was already set",
+                replicaThrottleRateConfigKey, brokerId, currThrottleRate);
+      }
+    }
+    if (!ops.isEmpty()) {
+      changeBrokerConfigs(brokerId, ops);
+    }
+  }
+
+  private Config getTopicConfigs(String topic) throws ExecutionException, InterruptedException {
+    try {
+      return getEntityConfigs(new ConfigResource(ConfigResource.Type.TOPIC, topic));
+    } catch (Exception e) {
+      if (!topicExists(topic)) {
+        return new Config(Collections.emptyList());
+      }
+      throw e;
     }
   }
 
-  private void setLeaderThrottledReplicas(String topic, Set<String> replicas) {
-    setThrottledReplicas(topic, replicas, true);
+  private Config getBrokerConfigs(int brokerId) throws ExecutionException, InterruptedException {
+    return getEntityConfigs(new ConfigResource(ConfigResource.Type.BROKER, String.valueOf(brokerId)));
   }
 
-  private void setFollowerThrottledReplicas(String topic, Set<String> replicas) {
-    setThrottledReplicas(topic, replicas, false);
+  private Config getEntityConfigs(ConfigResource cf) throws ExecutionException, InterruptedException {
+    Map<ConfigResource, Config> configs = _adminClient.describeConfigs(Collections.singletonList(cf)).all().get();
```

Comment:
Nit: Would it be possible to use `KafkaFuture#get(long timeout, TimeUnit unit)` rather than `KafkaFuture#get()`? This ensures that the request is blocked for at most the given time for this future to complete rather than waiting until the admin request times out.
Also applies to the other adminClient calls in this PR -- e.g. `incrementalAlterConfigs` and `listTopics`.